test/functional/tests/io/test_write_fetch.py

```diff
@@ -0,0 +1,210 @@
+            in cache statistics
+          - Appropriate number of writes to cache in iostat
+    """
+    pattern = f"0x{uuid.uuid4().hex}"
+    io_size = Size(600, Unit.MebiByte)
+
+    with TestRun.step("Prepare devices."):
+        cache_disk = TestRun.disks['cache']
+        core_disk = TestRun.disks['core']
+        core_disk.create_partitions([io_size + Size(1, Unit.MebiByte)])
+        core_part = core_disk.partitions[0]
+
+    with TestRun.step("Fill core partition with pattern."):
+        cache_mode_traits = CacheMode.get_traits(cache_mode)
+        if CacheModeTrait.InsertRead in cache_mode_traits:
+            run_fio(target=core_part.system_path,
+                    operation_type=ReadWrite.write,
+                    blocksize=Size(4, Unit.KibiByte),
+                    io_size=io_size,
+                    verify=True,
+                    pattern=pattern)
+        else:
+            TestRun.LOGGER.info(f"Skipped for {cache_mode} cache mode.")
+
+    with TestRun.step("Start cache and add core."):
+        cache = casadm.start_cache(cache_disk, cache_mode, cache_line_size)
+        Udev.disable()
+        core = cache.add_core(core_part)
+    with TestRun.step("Cache half of file."):
+        operation_type = ReadWrite.read if CacheModeTrait.InsertRead in cache_mode_traits \
+            else ReadWrite.write
+        run_fio(target=core.system_path,
+                operation_type=operation_type,
+                skip=cache_line_size.value,
+                blocksize=cache_line_size.value,
+                io_size=io_size,
+                verify=True,
+                pattern=pattern)
+        if CacheModeTrait.InsertRead not in cache_mode_traits:
+            cache.flush_cache()
+        casadm.reset_counters(cache.cache_id, core.core_id)
+    with TestRun.step("Run writes to CAS device using fio."):
+        io_stats_before_io = cache_disk.get_io_stats()
+        run_fio(target=core.system_path,
+                operation_type=ReadWrite.write,
+                skip=cache_line_size.value / 2,
+                blocksize=cache_line_size.value / 2 * 3,
+                io_size=io_size)
+    with TestRun.step("Verify CAS statistics for partial misses, write hits and writes to cache."):
+        check_statistics(cache=cache,
+                         io_interval=cache_line_size.value * 2,
+                         io_size=io_size,
+                         partial_misses=True)
+    with TestRun.step("Verify number of writes to cache device using iostat. Shall be 450M "
+                      "(+ metadata for cache mode with write insert feature)."):
+        check_io_stats(cache_disk=cache_disk,
+                       cache=cache,
+                       io_stats_before=io_stats_before_io,
+                       io_size=io_size,
+                       partial_misses=True)
+
+
+# Methods used in tests:
+def check_io_stats(cache_disk, cache, io_stats_before, io_size, partial_misses=False):
```

Comment:
Similarly to `check_statistics()` i would pass block_size and skip_size to make `expected_writes_mib` more clear